egs/TAC/train.py

```diff
@@ -0,0 +1,156 @@
+    # Define scheduler
+
+    if conf["training"]["half_lr"]:
+        scheduler = ReduceLROnPlateau(
+            optimizer=optimizer, factor=0.5, patience=conf["training"]["patience"]
+        )
+    else:
+        scheduler = None
+    # Just after instantiating, save the args. Easy loading in the future.
+    exp_dir = conf["main_args"]["exp_dir"]
+    os.makedirs(exp_dir, exist_ok=True)
+    conf_path = os.path.join(exp_dir, "conf.yml")
+    with open(conf_path, "w") as outfile:
+        yaml.safe_dump(conf, outfile)
+
+    # Define Loss function.
+    loss_func = PITLossWrapper(pairwise_neg_sisdr)
+    system = TACSystem(
+        model=model,
+        loss_func=loss_func,
+        optimizer=optimizer,
+        train_loader=train_loader,
+        val_loader=val_loader,
+        scheduler=scheduler,
+        config=conf,
+    )
+
+    # Define callbacks
+    checkpoint_dir = os.path.join(exp_dir, "checkpoints/")
+    checkpoint = ModelCheckpoint(
+        checkpoint_dir,
+        monitor="val_loss",
+        mode="min",
+        save_top_k=conf["training"]["save_top_k"],
+        verbose=True,
+    )
+    early_stopping = False
+    if conf["training"]["early_stop"]:
+        early_stopping = EarlyStopping(
+            monitor="val_loss", patience=conf["training"]["patience"], verbose=True
+        )
+
+    # Don't ask GPU if they are not available.
+    gpus = -1 if torch.cuda.is_available() else None
+    trainer = pl.Trainer(
+        max_epochs=conf["training"]["epochs"],
+        checkpoint_callback=checkpoint,
+        callbacks=[early_stopping],
+        default_root_dir=exp_dir,
+        gpus=gpus,
+        distributed_backend="ddp",
+        gradient_clip_val=conf["training"]["gradient_clipping"],
+    )
+    trainer.fit(system)
+
+    best_k = {k: v.item() for k, v in checkpoint.best_k_models.items()}
+    with open(os.path.join(exp_dir, "best_k_models.json"), "w") as f:
+        json.dump(best_k, f, indent=0)
+
+    state_dict = torch.load(checkpoint.best_model_path)
+    system.load_state_dict(state_dict=state_dict["state_dict"])
+    system.cpu()
+
+    to_save = system.model.serialize()
+    to_save.update(train_set.get_infos())
+    torch.save(to_save, os.path.join(exp_dir, "best_model.pth"))
+    save_publishable(
+        os.path.join(exp_dir, "publish_dir"),
+        to_save,
+        metrics=dict(),
+        train_conf=conf,
+        recipe="asteroid/TAC",
+    )
+
+
+if __name__ == "__main__":
+    import yaml
+    from pprint import pprint as print
+    from asteroid.utils import prepare_parser_from_dict, parse_args_as_dict
+
+    # We start with opening the config file conf.yml as a dictionary from
+    # which we can create parsers. Each top level key in the dictionary defined
+    # by the YAML file creates a group in the parser.
+    with open("./local/conf.yml") as f:
+        def_conf = yaml.safe_load(f)
+    parser = prepare_parser_from_dict(def_conf, parser=parser)
```

Comment:
I misunderstood something. 
I thought that you wanted to pass config files directly, instead of the usual way, and that the conf file could be passed in run.sh?